haskell/test/Test/Plist/Property.hs

```diff
@@ -0,0 +1,128 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.Plist.Property
+  ( hprop_Plist
+  ) where
+
+import Universum hiding (drop, swap)
+
+import qualified Data.List as DL
+import Data.List ((!!))
+import Fmt (build, unlinesF)
+import Hedgehog hiding (assert)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Lorentz hiding (assert, and, div, now, (>>))
+import Test.Cleveland
+
+import Ligo.BaseDAO.Types
+import SMT.Model.BaseDAO.Proposal.Plist
+import Test.Ligo.BaseDAO.Plist (genProposalKeyList)
+import Test.Plist.Type
+import Test.Plist.Contract
+
+
+genPlistParameter :: MonadGen m => [ProposalKey] -> [ProposalKey] -> m PlistParameter
+genPlistParameter keyList keyList2 = do
+  (i :: Int) <- Gen.integral (Range.constant 0 (length keyList - 1))
+  let k = keyList !! i
+  (i2 :: Int) <- Gen.integral (Range.constant 0 (length keyList2 - 1))
+  let k2 = keyList2 !! i2
+  Gen.choice
+    [ pure $ Insert k2
+    , pure $ Mem k2
+    , pure $ Mem k
+    , pure $ Delete k
+    , pure $ Pop ()
+    ]
+
+
+genPlistParameters :: MonadGen m => m ([ProposalKey], [PlistParameter])
+genPlistParameters = do
+  initSize <- Gen.integral (Range.constant 1 10)
```

Comment:
Shouldn't range start from `0`?  Shouldn't we also include empty lists in the generated possibilites?